apps/passport-server/test/lemonade/LemonadeDataMocker.ts

```diff
@@ -1,151 +1,255 @@
-
-    if (!event) {
-      throw new Error(`can't add ticket to event that doesn't exist`);
+    if (!this.events.has(eventId)) {
+      throw new Error(`Can't add ticket to non-existent event ${eventId}`);
     }
+    if (!this.ticketTypes.get(eventId)?.has(ticketTypeId)) {
+      throw new Error(
+        `Can't add ticket of non-existent type ${ticketTypeId} to event ${eventId}`
+      );
+    }
+    if (!this.users.has(userId)) {
+      throw new Error(`Can't add ticket for non-existent user ${userId}`);
+    }
+    const ticket: LemonadeTicket = {
+      _id: randomUUID(),
+      type_id: ticketTypeId,
+      type_title: this.ticketTypes.get(eventId)?.get(ticketTypeId)
+        ?.title as string,
+      user_id: userId,
+      user_email: this.users.get(userId)?.email as string,
+      user_name: userName,
+      user_first_name: "",
+      user_last_name: "",
+      assigned_to: userId,
+      assigned_email: "", // This is what we get from the live API, if you
+      // want the email address then look in `user_email`
+      checkin_date: null
+    };
 
-    const tier = this.getTier(eventId, tierId);
+    const eventTickets = this.tickets.get(eventId) ?? new Map();
+    eventTickets.set(ticket._id, ticket);
+    this.tickets.set(eventId, eventTickets);
 
-    if (!tier) {
-      throw new Error(`can't add ticket to tier that doesn't exist`);
-    }
+    return ticket;
+  }
 
-    event.tickets.push(newTicket);
+  public getEvents(): Map<string, LemonadeEvent> {
+    return this.events;
+  }
 
-    return newTicket;
+  public getTicketTypes(): Map<string, Map<string, LemonadeTicketType>> {
+    return this.ticketTypes;
   }
 
-  public addEvent(name: string): LemonadeEvent {
-    const newEvent: LemonadeEvent = {
-      id: randomUUID(),
-      name: name,
-      tickets: [],
-      tiers: [],
-      permissionedUserIds: []
-    };
-    this.events.push(newEvent);
-    return newEvent;
+  public getTickets(): Map<string, Map<string, LemonadeTicket>> {
+    return this.tickets;
   }
 
-  public addTier(eventId: string, name: string): LemonadeTicketTier {
-    const newTier: LemonadeTicketTier = {
-      id: randomUUID(),
-      name
-    };
-    const event = this.events.find((e) => e.id === eventId);
-    if (!event) {
-      throw new Error(`unable to find event with id ${eventId}`);
+  /**
+   * Check in a user.
+   */
+  public setCheckin(
+    eventId: string,
+    userId: string,
+    checkinDate: Date | null
+  ): void {
+    if (!this.events.has(eventId)) {
+      throw new Error(`Can't check in user to non-existent event ${eventId}`);
     }
-    const existingTier = event.tiers.find((t) => t.name === name);
-    if (existingTier) {
+    if (!this.users.has(userId)) {
+      throw new Error(`Can't check in non-existent user ${userId}`);
+    }
+
+    const tickets = this.getTickets().get(eventId)?.values();
+    let ticket;
+
+    if (
+      !tickets ||
+      !(ticket = [...tickets].find((t) => t.assigned_to === userId))
```

Comment:
can we move this assignment to the `ticket` variable to line 170?

Reply:
Then it will complain that `tickets` might be undefined.